### PR TITLE
Enhancement controller as extern for online test

### PR DESCRIPTION
**Pull request by Jean-Eudes-le-retour**
This PR adds compatibility to a new dockerCompose type in a simulation's webots.yml specific to benchmarks in the form of  `dockerCompose: default_controller:pit_escape` for example

It launches the default controller inside its own docker container and therefore decouples the dependencies between the Webots container and the controller.

This would also make the online test more coherent with the GitHub runner evaluations as they will done in the same way.

It would also make the webots.yml edit for the default controller's name more intuitive for the participants as it would explicitly say "default_controller"

***

It can be tested online [here](https://benchmark.webots.cloud/run?version=R2023a&url=https%3A%2F%2Fgithub.com%2FJean-Eudes-le-retour%2FPit-Escape-benchmark%2Fblob%2Fmain%2Fworlds%2Fpit_escape.wbt&type=benchmark&context=try)

### CODE REVIEW

File: config/simulation/docker/remote_controller_launcher.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import importlib.util
+import os
+from controller.wb import wb
+
+
+class Initializer:
+    def __init__(self):
+        # Running wb_robot_init() before controller execution to send the logs to webots container stdout even if controller crashes.
```
Comment (omichel) [inline config/simulation/docker/remote_controller_launcher.py]:
```suggestion
        # Running wb_robot_init() before controller execution to send the logs to webots container stdout even if the controller crashes.
```

File: simulation_server.py [hunk 2]
```diff
@@ -324,6 +324,16 @@ def runWebotsInThread(client):
...
                                     client.websocket.write_message('ide: enable')
+                                elif info[1].strip() == 'default_controller':
+                                    default_controller = info[2]
+                                    dockerComposePath = config['dockerConfDir'] + "/docker-compose-benchmark.yml"
+                                    envVarDocker["DEFAULT_CONTROLLER"] = default_controller
+                                    envVarDocker["THEIA_PORT"] = port + 500
+                                    client.websocket.write_message('ide: enable')
+                                    os.system(
+                                        f'cp {config["dockerConfDir"]}/remote_controller_launcher.py '
+                                        f'./remote_controller_launcher.py'
```
Comment (omichel) [inline simulation_server.py]: Wouldn't this be simpler?
```suggestion
                                        f'ln -s {config["dockerConfDir"]}/remote_controller_launcher.py'
```

**Review by omichel — CHANGES_REQUESTED** [review]
That looks good to me.
I just have a couple of suggestions.

File: simulation_server.py (hunk 2 above)
Comment (Jean-Eudes-le-retour) [reply simulation_server.py]: Apparently during a Dockerfile build, COPY instructions can't resolve simlinks outside the build context so I have to make a real file copy

Comment (omichel) [reply simulation_server.py]: OK, so let's revert to the copy and add a comment explaining why we cannot make a soft link.

Comment (Jean-Eudes-le-retour) [reply simulation_server.py]: Apparently hard links work, I'll use that instead

**Review by omichel — APPROVED** [review]
Looks good to me.
Thank you.

**Pull request merged**